### PR TITLE
feat: toggle on/off

### LINKED ISSUE #2: Allow toggling on/off
Thanks for creating this!  Is there a command that we could use to toggle on/off that we could `map` to?

### PR DESCRIPTION
**Pull request by zxt-tzx**
Closes https://github.com/nadavspi/obsidian-relative-line-numbers/issues/2

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -28,15 +46,19 @@ export default class RelativeLineNumbers extends Plugin {
...
+      this.editorExtension.length = 0;
+      this.editorExtension.push(lineNumbersRelative());
+      this.app.workspace.updateOptions();
     }
   }
 
   disable() {
     this.enabled = false;
-    if (this.isLegacy) {
+    if (this.isLegacy()) {
```
Comment (zxt-tzx) [inline main.ts:57]: bug from original code — this is always truthy

File: package.json [hunk 2]
```diff
@@ -12,16 +12,16 @@
...
+    "@codemirror/language": "^6.6.0",
+    "@codemirror/state": "^6.2.0",
+    "@codemirror/view": "^6.2.0",
     "@types/node": "^16.11.6",
     "@typescript-eslint/eslint-plugin": "^5.2.0",
     "@typescript-eslint/parser": "^5.2.0",
     "builtin-modules": "^3.2.0",
     "esbuild": "0.13.12",
-    "obsidian": "^0.12.17",
+    "obsidian": "1.4.11",
```
Comment (zxt-tzx) [inline package.json:23]: updating this removes some type errors

File: package.json [hunk 3]
```diff
@@ -12,16 +12,16 @@
...
     "tslib": "2.3.1",
-    "typescript": "4.4.4",
-    "@codemirror/state": "^6.2.0",
-    "@codemirror/view": "^6.2.0",
-    "@codemirror/language": "^6.6.0"
+    "typescript": "^5.3.3"
```
Comment (zxt-tzx) [inline package.json:25]: could not build using TS 4.4.4

File: main.ts [hunk 4]
```diff
@@ -28,15 +44,19 @@ export default class RelativeLineNumbers extends Plugin {
     if (this.isLegacy()) {
       this.legacyEnable();
     } else {
-      // @ts-ignore
-      this.registerEditorExtension(lineNumbersRelative());
+      this.editorExtension.length = 0;
```
Comment (zxt-tzx) [inline main.ts:47]: this follows code style from these guidelines https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Change+or+reconfigure+editor+extensions

**Comment by nadavspi** [comment]
Nice, thanks for this

**Pull request merged**